src/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -558,10 +561,20 @@ public static void makeReadUnmapped(final SAMRecord rec) {
         rec.setMappingQuality(SAMRecord.NO_MAPPING_QUALITY);
         rec.setInferredInsertSize(0);
         rec.setNotPrimaryAlignmentFlag(false);
+        rec.setSupplementaryAlignmentFlag(false);
```

Comment:
This seems wrong. it would lead to out-of-spec bamfiles that have two primary alignments with the same name. the previous line seems similarly wrong (I know you didn't change it). I'll open an issue and see what people think.